q2_moshpit/prodigal/prodigal.py

```diff
@@ -24,34 +26,56 @@ def predict_genes_prodigal(
     genes = GenesDirectoryFormat()
     proteins = ProteinsDirectoryFormat()
 
-    # Get paths to fasta files in input dir
-    fasta_files = [
-        file for file in os.listdir(mags.path)
-        if file.endswith(".fa") or file.endswith(".fasta")
-    ]
-
     # Define base command
     base_cmd = [
         "prodigal",
         "-g", translation_table_number,
         "-f", "gff"
     ]
 
-    # For every fasta file in mags.path call prodigal and write
-    # outputs to the corresponding directories.
-    for fasta_file in fasta_files:
-        # Get the filename from the file path
-        file_id = os.path.splitext(fasta_file)[0]
-
-        # Adjust command and run
-        cmd = cp.deepcopy(base_cmd)
-        cmd.extend([
-            "-i", os.path.join(mags.path, fasta_file),
-            "-o", os.path.join(loci.path, f"{file_id}_loci.gff"),
-            "-a", os.path.join(proteins.path, f"{file_id}_proteins.fasta"),
-            "-d", os.path.join(genes.path, f"{file_id}_genes.fasta")
-        ])
-        run_command(cmd)
+    def _process_fasta_files(fasta_files: list, prefix: str, input_path: str):
+        # For every fasta file call prodigal and write
+        # outputs to the corresponding directories.
+        for fasta_file in fasta_files:
+            # Get the filename from the file path
+            file_id = os.path.splitext(fasta_file)[0]
+
+            # Adjust command and run
+            cmd = cp.deepcopy(base_cmd)
+            cmd.extend([
+                "-i", os.path.join(input_path, fasta_file),
+                "-o",
+                os.path.join(
+                    loci.path, f"{prefix}{file_id}_loci.gff"
+                ),
+                "-a",
+                os.path.join(
+                    proteins.path, f"{prefix}{file_id}_proteins.fasta"
+                ),
+                "-d",
+                os.path.join(
+                    genes.path, f"{prefix}{file_id}_genes.fasta"
+                )
+            ])
+            run_command(cmd)
+
+    if isinstance(mags, MAGSequencesDirFmt):
+        # Get paths to fasta files in input dir
+        fasta_files = os.listdir(mags.path)
+        _process_fasta_files(fasta_files, '', mags.path)
+
+    elif isinstance(mags, MultiMAGSequencesDirFmt):
+        # List all directories / samples
+        for sample_dir in os.listdir(mags.path):
+            sample_dir_path = os.path.join(mags.path, sample_dir)
+            if os.path.isdir(sample_dir_path):
+                fasta_files = os.listdir(sample_dir_path)
+                print(fasta_files)
+                _process_fasta_files(
```

Comment:
This can then also be rewritten like the loop above, just that here you should use the `sample_dict` method from the `MultiMAGSequencesDirFmt` (https://github.com/qiime2/q2-types/blob/d24a69d649c98307851bee01b8c61a684ad864d1/q2_types/per_sample_sequences/_format.py#L608) to get all the IDs and paths.